### PR TITLE
fix: dont throw when dot is null, use typeguard in line and improve in area

### PR DESCRIPTION
**Pull request by ckifer**
## Description

When the `clipDots` PR was merged I added a typeguard to `Area` but I did not see to do the same for `Line`. If `dot` was something other than `DotProps` this cast is incorrect and wrong. 

Linked issue is caused when passing `null` to `dot`.  `null` is not an acceptable type to `dot` and therefore this wasn't a breaking change, but it still uncovered a need for this check.

## Related Issue

fixes https://github.com/recharts/recharts/issues/3635

## Motivation and Context

See issue, removes need to cast to `DotProps`

## How Has This Been Tested?

Added unit tests that reproduce the issue and are green after fixing it

## Screenshots (if appropriate):

N/A no visual changes

## Types of changes

- [Y] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [Y] My code follows the code style of this project.
- [N] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [Y] I have added tests to cover my changes.
- [Y] All new and existing tests passed.

### CODE REVIEW

**Comment by ckifer** [comment]
going to merge this to get a release out before I'm on vacation for a bit. Please review in retrospect for red-flags.

File: src/util/ReactUtils.ts [hunk 1]
```diff
@@ -260,6 +262,9 @@ const SVG_TAGS: string[] = [
 
 const isSvgElement = (child: any) => child && child.type && _.isString(child.type) && SVG_TAGS.indexOf(child.type) >= 0;
 
+export const isDotProps = (dot: LineDot | AreaDot): dot is DotProps =>
+  dot && typeof dot === 'object' && 'cx' in dot && 'cy' in dot && 'r' in dot;
```
Comment (nikolasrieble) [inline src/util/ReactUtils.ts:266]: Nit: It might be nice going forward to start smaller files with smaller scope to move away from generic collect all files. 

This function for example could live in a separate file. It seems unrelated to all other functions. 

But that really is just personal preference.

**Review by nikolasrieble — COMMENTED** [review]
👍 I see no issue, thank you for your quick reaction!

File: src/util/ReactUtils.ts (hunk 1 above)
Comment (ckifer) [reply src/util/ReactUtils.ts:266]: Sure - to me at the time it seemed meta enough that 'ReactUtil' fit, but in reality its more of a TypeScript util or even a specific Dot util
